AgentAssistAngular/src/app/services/template-render-class.service.ts

```diff
@@ -159,7 +160,20 @@ export class TemplateRenderClassService {
       };
 
     } else {
-      body['component'] = parsedPayload.payload ? parsedPayload : parsedPayload.text;
+
+      // body['component'] = parsedPayload.payload ? parsedPayload : parsedPayload.text;
+
+      if (parsedPayload.payload) {
+        body.component = parsedPayload;
+      } else if (parsedPayload.text) {
```

Comment:
@saradadh-Kore Please modify according to below for better Readability and avoiding multiple dot operations.

        body.component = {
			payload : {
	          text: parsedPayload,
			},
			type = 'text'
        }